crates/jxl-render/src/lib.rs

```diff
@@ -385,3 +385,30 @@ impl<'a> std::ops::DerefMut for IndexedFrame<'a> {
         &mut self.f
     }
 }
+
+pub fn render_spot_colour(
+    grid: &mut [SimpleGrid<f32>],
+    ec_grid: &SimpleGrid<f32>,
+    ec_ty: &ExtraChannelType,
+) -> Result<()> {
+    let ExtraChannelType::SpotColour { red, green, blue, solidity } = ec_ty else {
+        return Err(Error::NotSupported("not a spot colour ec"));
+    };
+
+    let spot_colors = [red, green, blue];
+    let s = ec_grid.buf();
+    let width = ec_grid.width();
+    let height = ec_grid.height();
+
+    (0..grid.len()).for_each(|c| {
+        let buf = grid[c].buf_mut();
+        let color = spot_colors[c];
+        (0..height).for_each(|y| {
+            (0..width).for_each(|x| {
+                let mix = s[y * width + x] * solidity;
+                buf[y * width + x] = mix * color + (1.0 - mix) * buf[y * width + x];
+            });
+        });
```

Comment:
Or, you can iterate over `grid` and `ec_grid` at the same time, using the single index in range `0..grid.buf().len()` (assuming all grid sizes are the same.) This might help auto-vectorization.

Reply:
Thanks!